app/src/main/java/com/pileproject/drive/util/bluetooth/RxBluetoothConnector.java

```diff
@@ -77,11 +81,47 @@ public void call(Subscriber<? super BluetoothSocket> subscriber) {
 
                     subscriber.onNext(socket);
                     subscriber.onCompleted();
-                } catch (IOException e) {
-                    subscriber.onError(e);
+                } catch (IOException firstCause) {
+
+                    try {
+                        BluetoothSocket socket = connectByRfcommSocket(bluetoothDevice);
+                        subscriber.onNext(socket);
+                    } catch (RfcommSocketInvocationException e) {
+                        subscriber.onError(firstCause);
```

Comment:
You may as well mention which method (first or second) failed to connect. Users can't see which line (this or L. 92) pass the exception to the subscriber.